File: 09-DesignPatterns-Java/Solution/Observer/src/ConsoleView.java
public class ConsoleView implements IView {
    private final int _id;

    public ConsoleView(int id) {
        _id = id;
    }

    public void update(float temperature, float humidity, float windSpeed, float rainProbability) {
        System.out.println("View #" + _id + " | Temperature: " + temperature + " C° | Humidity: " + humidity + " % | Windspeed: " + windSpeed + " m/s | Rain Probability: " + rainProbability + "%");
    }
}

File: 10-DesignPatterns-CSharp/10-1-1-Observer-Solution/ConsoleView.cs
namespace Observer
{
    using System;

    public sealed class ConsoleView : IView
    {
        private readonly int _id;

        public ConsoleView(int id)
        {
            _id = id;
        }

        public void Update(float temperature, float humidity, float windSpeed, float rainProbability)
        {
            Console.WriteLine($"View #{_id} | Temperature: {temperature} C° | Humidity: {humidity} % | Windspeed: {windSpeed} m/s | Rain Probability: {rainProbability} %");
        }
    }
}
